packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/features/sensororientation/DeviceOrientationManager.java

```diff
@@ -0,0 +1,214 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.camera.features.sensororientation;
+
+import android.app.Activity;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.content.res.Configuration;
+import android.hardware.SensorManager;
+import android.provider.Settings;
+import android.view.Display;
+import android.view.OrientationEventListener;
+import android.view.Surface;
+import android.view.WindowManager;
+import androidx.annotation.NonNull;
+import io.flutter.embedding.engine.systemchannels.PlatformChannel;
+import io.flutter.plugins.camera.DartMessenger;
+
+public class DeviceOrientationManager {
+
+  private static final IntentFilter orientationIntentFilter =
+      new IntentFilter(Intent.ACTION_CONFIGURATION_CHANGED);
+
+  private final Activity activity;
+  private final DartMessenger messenger;
+  private final boolean isFrontFacing;
+  private final int sensorOrientation;
+  private PlatformChannel.DeviceOrientation lastOrientation;
+  private OrientationEventListener orientationEventListener;
+  private BroadcastReceiver broadcastReceiver;
+
+  /** Factory method to create a device orientation manager. */
+  public static DeviceOrientationManager create(
+      @NonNull Activity activity,
+      @NonNull DartMessenger messenger,
+      boolean isFrontFacing,
+      int sensorOrientation) {
+    return new DeviceOrientationManager(activity, messenger, isFrontFacing, sensorOrientation);
+  }
+
+  private DeviceOrientationManager(
+      @NonNull Activity activity,
+      @NonNull DartMessenger messenger,
+      boolean isFrontFacing,
+      int sensorOrientation) {
+    this.activity = activity;
+    this.messenger = messenger;
+    this.isFrontFacing = isFrontFacing;
+    this.sensorOrientation = sensorOrientation;
+  }
+
+  public void start() {
+    startSensorListener();
```

Comment:
Could you add a comment here explaining why we listen to both? It's not clear to me why we need to compute the orientation two different ways. Is it a case of needing a fallback if one isn't available?

Reply:
Done.